src/main/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/CaseServiceImpl.java

```diff
@@ -63,6 +69,43 @@ public CaseDTO getLatestValidNonHICaseByUPRN(final UniquePropertyReferenceNumber
     }
   }
 
+  @Override
+  public CaseDTO createNewCase(CaseRequestDTO request) throws CTPException {
+
+    Optional<CaseDTO> existingCase = getLatestCaseByUPRN(request.getUprn());
+
+    CaseDTO caseToReturn;
+    if (existingCase.isPresent()) {
+      // Don't need to create a new case, as we found one with the same UPRN
+      caseToReturn = existingCase.get();
+    } else {
+      // Create a new case as not found for the UPRN in Firestore
+      CaseType caseType = determineCaseType(request);
+      CollectionCase newCase =
+          ServiceUtil.createCase(request, caseType, appConfig.getCollectionExerciseId());
+      log.with("caseId", newCase.getId())
+          .with("primaryCaseType", caseType)
+          .debug("Created new case");
+
+      // Store new case in Firestore
+      dataRepo.writeCollectionCase(newCase);
+
+      // tell RM we have created a case for the selected (HH|CE|SPG) address
+      ServiceUtil.sendNewAddressEvent(eventPublisher, newCase);
+
+      caseToReturn = mapperFacade.map(newCase, CaseDTO.class);
+    }
+
+    // Set address level for case
```

Comment:
This code looks similar to **determineAddressLevel** method in **UniqueAccessCodeServiceImpl** . Might be possible to put a method in **ServiceUtil** for reuse. Also should SPG type be considered in the same way ?

Reply:
I've discussed this statement in detail with Phil, who has then had a discussion with Rob Smart. They think that it should just be CE cases which will have the address level set to E. 
This code is indeed very similar to determineAddressLevel(), and if was then the same then there should be a common method, but the differences mean that I don't think it makes sense to merge them into one method. It would have to take a UAC (or null) and some sort of mode enum to control which if statement gets used. The medicine would be worse than the illness.

Reply:
ok